Lib/configparser.py

```diff
@@ -1106,10 +1106,12 @@ def _handle_rest(self, st, line, fpname):
         # a section header or option header?
         if self._allow_unnamed_section and st.cursect is None:
             st.sectname = UNNAMED_SECTION
-            st.cursect = self._dict()
-            self._sections[st.sectname] = st.cursect
-            self._proxies[st.sectname] = SectionProxy(self, st.sectname)
-            st.elements_added.add(st.sectname)
```

Comment:
"st.elements_added.add(st.sectname)" is removed because it is responsible for checking for section duplication. But an unnamed section cannot be duplicated within a file. When reading the next file, the parser is reset to its initial state and the elements_added set is cleared. Moreover, there was no duplication check in the original code, only adding to the set. The remaining lines have been moved under the "else" statement since they should only be executed when the section is created.